### PR TITLE
Apply pixel scaling to images in the viewer and add an Image/Labels writer

### PR DESCRIPTION
**Pull request by p-j-smith**
- the scaling and translation metadata read from tiff files is used to scale images / labels layers in the viewer

- I've also added an EpiTools writer so that output from the various steps (projection, segmentation) can be exported and loaded again

- I've added a module `napari_epitools.widgets.dialogue` with a function `select_option`. This is a general function for creating a dialogour box with a dropdown list of options from which a user should select one. It's used here when saving a file. If an image has e.g 4 dimensions, the axes dimensions could represent a:
    - three dimensional timeseries (TZYX)
    - three dimensional multichannel image (ZYXC)
    - two dimensional multichannel timeseries (TYXC)

so the dialogue box is used to ask the user for the correct axes dimensions. Here's what the dialogue pop up looks like:

epitools-save-dialogue

- I've added a test for the dialogue box based on test for a similar function in magicgui: https://github.com/pyapp-kit/magicgui/blob/main/tests/test_widgets.py#L918

- I've had to undo the changes made in #40, i.e. images and labels layers now all have a `Z` dimension, even if it is a single slice. This is because `PartSeg` always adds the `Z` dimension into a `TYX` image when loading it, and there was no easy way around this. Unfortunately this the output from epitools now doesn't work with the btrack plugin. However, btrack itself can handle 4d data. To make the plugin able to handle 4d data, we'd need to change [this line](https://github.com/quantumjot/btrack/blob/main/btrack/napari/main.py#L173):

```python
data, properties, graph = tracker.to_napari(ndim=2)
```
to:
```python
data, properties, graph = tracker.to_napari()
```

which would let `btrack` determine the dimensionality of the layer

### CODE REVIEW

File: src/napari_epitools/analysis/projection.py [hunk 1]
```diff
@@ -120,7 +121,9 @@ def calculate_projection(
     t_size, z_size, y_size, x_size = input_image.shape
     smoothed_imstack = _smooth(input_image.astype(np.float64), smoothing_radius)
 
-    t_interp = np.zeros((t_size, y_size, x_size))  # remove the z-axis when projecting
+    t_interp = np.zeros(
+        (t_size, SINGLE_SLICE, y_size, x_size)
+    )  # single slice in the Z dimension
```
Comment (paddyroddy) [inline src/napari_epitools/analysis/projection.py:126]: More readable
```suggestion
    # single slice in the Z dimension
    t_interp = np.zeros((t_size, SINGLE_SLICE, y_size, x_size))
```

File: src/napari_epitools/_writer.py [hunk 2]
```diff
@@ -8,17 +8,147 @@
...
+
+    selected = napari_epitools.widgets.dialogue.select_option(
+        options=options.keys(),
+        title=title,
+        prompt=prompt,
+        parent=parent,
+    )
+    axes_order = selected if selected is None else options[selected]
+
+    return axes_order
```
Comment (paddyroddy) [inline src/napari_epitools/_writer.py:80]:
```suggestion
    return selected if selected is None else options[selected]
```

File: src/napari_epitools/_reader.py [hunk 3]
```diff
@@ -81,12 +81,8 @@ def reader_function(path):
...
         layer_kwargs["metadata"] = {
-            "scale": np.asarray(
-                layer_kwargs.pop("scale")
-            ),  # don't scale the image in the viewer
-            "spacing": np.asarray(
-                image_stack.image.spacing
-            ),  # we need this for regionprops
+            "yx_spacing": np.asarray(
+                image_stack.image.spacing[1:]
+            ),  # ZYX[1:], we need this spacing for regionprops
```
Comment (paddyroddy) [inline src/napari_epitools/_reader.py:86]: More readable
```suggestion
            # ZYX[1:], we need this spacing for regionprops
            "yx_spacing": np.asarray(image_stack.image.spacing[1:]),
```

File: src/napari_epitools/widgets/dialogue.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        An optional parent widget, by default None.
+        The dialog will inherit style from the parent object.
+        CAREFUL: if a parent is set, and subsequently deleted, this widget will likely
+        be deleted as well (depending on the backend), and will no longer be usable.
+
+    Returns
+    -------
+    Optional[Any]
+        Selected value if accepted, or ``None`` if cancelled.
+
```
Comment (paddyroddy) [inline src/napari_epitools/widgets/dialogue.py:46]: Any reason?
```suggestion

```

**Comment by p-j-smith** [comment]
I'll finish adding some tests in #71 before wrapping this one up, just so we can check this pr isn't breaking anything

File: src/napari_epitools/analysis/projection.py (hunk 1 above)
Comment (p-j-smith) [reply src/napari_epitools/analysis/projection.py]: I've moved the commend to the top of the block and expanded it a bit

File: src/napari_epitools/_reader.py (hunk 3 above)
Comment (p-j-smith) [reply src/napari_epitools/_reader.py]: I've moved the commend to the top of the block and expanded it a bit

File: src/napari_epitools/widgets/dialogue.py (hunk 4 above)
Comment (p-j-smith) [reply src/napari_epitools/widgets/dialogue.py]: not really, just habit, I guess I think it looks a little nicer to have it spaced out. But I've removed the empty line

**Review by paddyroddy — APPROVED** [review]
Good stuff

**Comment by paddyroddy** [comment]
Merging in case 👶

**Pull request merged**